src/GraphBLAS-sharp.Backend/Matrix/SpGeMM/Expand.fs

```diff
@@ -181,21 +181,31 @@ module Expand =
 
         fun (processor: MailboxProcessor<_>) (values: ClArray<'a>) (columns: ClArray<int>) (rows: ClArray<int>) ->
             // sort by columns
-            let valuesSortedByColumns =
+            let keys, valuesSortedByColumns =
```

Comment:
Do you think it's a good idea to add keys explicitly? In my opinion, it used to be more convenient to use sorting. Maybe we need two separate implementations?